libbeat/publisher/queue/diskqueue/segments.go

```diff
@@ -354,36 +407,20 @@ func readSegmentHeader(in io.Reader) (*segmentHeader, error) {
 	if err != nil {
 		return nil, err
 	}
-	if header.version > currentSegmentVersion {
-		return nil, fmt.Errorf("unrecognized schema version %d", header.version)
-	}
-	if header.version >= 1 {
+	switch header.version {
+	case 0:
```

Comment:
Just to improve readability and make it explicit that there is no `fallthrough` behaviour expected in this switch.
```suggestion
	case 0: // do nothing
```